### PR TITLE
Task 89816 map retry sleep durations from Entity to DirectorService

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.Domain/Entities/WebhooksEntity.cs [hunk 1]
```diff
@@ -112,6 +127,7 @@ public OperationResult<WebhooksEntity> SetHooks(WebhooksEntity webhooks, Subscri
             SetSelectionRule(webhooks.SelectionRule);
             SetUriTransform(webhooks.UriTransform);
             SetPayloadTransform(webhooks.PayloadTransform);
+            SetRetrySleepDurationTransform(webhooks.RetrySleepDurations);
```
Comment (mszymeshop) [inline src/CaptainHook.Domain/Entities/WebhooksEntity.cs:130]: Why does it say `Transform` in the name of the method?

File: src/Tests/CaptainHook.Application.Tests/Infrastructure/SubscriberEntityToConfigurationMapperTests/WebhookAndCallbacksMapSubscriberAsyncTests.cs [hunk 2]
```diff
@@ -680,8 +788,8 @@ public async Task WhenDifferentPayloadTransformsAreUsed_MapToCorrectRule(string
             var result = await new SubscriberEntityToConfigurationMapper(_secretProviderMock.Object).MapToWebhookAsync(subscriber);
 
             // Assert
-            using(new AssertionScope())
-            result.IsError.Should().BeFalse();
+            using (new AssertionScope())
```
Comment (mszymeshop) [inline src/Tests/CaptainHook.Application.Tests/Infrastructure/SubscriberEntityToConfigurationMapperTests/WebhookAndCallbacksMapSubscriberAsyncTests.cs:791]: I think this is incorrect in general, it should be either in `{}` or `using var _ = new AsertionScope()`. At the moment the scope will apply to the first line only.

File: src/CaptainHook.Application.Infrastructure/Mappers/SubscriberEntityToConfigurationMapper.cs [hunk 3]
```diff
@@ -157,14 +157,16 @@ private async Task<OperationResult<WebhookConfig>> MapSingleWebhookWithNoUriTran
...
+            var config = new WebhookConfig
             {
                 Name = name,
                 EventType = eventType,
                 Uri = webhooksEntity?.Endpoints?.FirstOrDefault()?.Uri,
                 HttpVerb = webhooksEntity?.Endpoints?.FirstOrDefault()?.HttpVerb,
                 AuthenticationConfig = authenticationResult.Data
             };
+
+            return SetRetrySleepDurationIfOverriden(config, webhooksEntity);
```
Comment (tabman83) [inline src/CaptainHook.Application.Infrastructure/Mappers/SubscriberEntityToConfigurationMapper.cs:169]: overridden

**Review by mszymeshop — APPROVED** [review]

File: src/CaptainHook.Domain/Entities/WebhooksEntity.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.Domain/Entities/WebhooksEntity.cs]: Fixed

File: src/Tests/CaptainHook.Application.Tests/Infrastructure/SubscriberEntityToConfigurationMapperTests/WebhookAndCallbacksMapSubscriberAsyncTests.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Application.Tests/Infrastructure/SubscriberEntityToConfigurationMapperTests/WebhookAndCallbacksMapSubscriberAsyncTests.cs:791]: Fixed

**Pull request merged**